[tune] Introduce `durable()` wrapper to convert trainables into durable trainables (#14306)
ray-project/ray: doc/source/tune/api_docs/trainable.rst

.. autofunction:: ray.tune.integration.tensorflow.DistributedTrainableCreator
   :noindex:

.. _tune-durable-trainable:

tune.DurableTrainable
---------------------

> The :class:`DurableTrainable <ray.tune.DurableTrainable>` can be used to sync trial logs and checkpoints to the cloud. This is especially
> useful when training a large number of distributed trials, as logs and checkpoints are otherwise synchronized
> via SSH, which quickly can become a performance bottleneck.
> 
> The :class:`DurableTrainable <ray.tune.DurableTrainable>` class inherits from
> :class:`Trainable <ray.tune.Trainable>` and thus can be extended like the base class. There's also an
> easy to use wrapper function, :func:`ray.tune.durable` that can be used to convert any kind of trainable
> to a ``DurableTrainable``, including pre-registered RLLib trainers and :ref:`function trainables <tune-function-api>`.

Reviewer: ```suggestion

Tune provides a :func:`ray.tune.durable` wrapper that can be used to convert any kind of trainable
to a ``DurableTrainable``, including pre-registered RLLib trainers and :ref:`function trainables <tune-function-api>`.

The :class:`DurableTrainable <ray.tune.DurableTrainable>` syncs trial logs and checkpoints to cloud storage (via the `upload_dir`). This is especially
useful when training a large number of distributed trials, as logs and checkpoints are otherwise synchronized
via SSH, which quickly can become a performance bottleneck. The :class:`DurableTrainable <ray.tune.DurableTrainable>` class inherits from
:class:`Trainable <ray.tune.Trainable>` and thus can be extended like the base class. 
```

.. autoclass:: ray.tune.DurableTrainable

> .. autofunction:: ray.tune.durable

Reviewer: can we add a paragraph for usage in the documentation?
